plugins/block-dynamic-connection/src/dynamic_text_join.js

```diff
@@ -0,0 +1,124 @@
+Blockly.Blocks['dynamic_text_join'] = {
+  inputCounter: 2,
+  /**
+   * Block for concatenating any number of strings.
+   * @this {Blockly.Block}
+   */
+  init: function() {
+    this.setHelpUrl(Blockly.Msg['TEXT_JOIN_HELPURL']);
+    this.setStyle('text_blocks');
+    this.appendValueInput('ADD0')
+        .appendField(Blockly.Msg['TEXT_JOIN_TITLE_CREATEWITH']);
+    this.appendValueInput('ADD1');
+    this.setOutput(true, 'String');
+    this.setTooltip(Blockly.Msg['TEXT_JOIN_TOOLTIP']);
+  },
+  /**
+   * Create XML to represent number of text inputs.
+   * @return {!Element} XML storage element.
+   * @this {Blockly.Block}
+   */
+  mutationToDom: function() {
+    const container = Blockly.utils.xml.createElement('mutation');
+    const inputNames = this.inputList.map((input) => input.name).join(',');
```

Comment:
Yeah we do, because the inputs won't always be ordered from 0. For example, if you make a block with four inputs, then remove the top two, the inputs will be named `ADD2` and `ADD3`. If we just stored the number, the next time we create the block we would name the inputs `ADD0` and `ADD1`, and the inputs wouldn't match.